src/main/java/com/workshop/demo/service/impl/ReviewServiceImpl.java

```diff
@@ -0,0 +1,123 @@
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import com.workshop.demo.exception.ResourceNotFoundException;
+import com.workshop.demo.exception.BlogapiException;
+import com.workshop.demo.model.Restaurant;
+import com.workshop.demo.model.Review;
+import com.workshop.demo.model.User;
+import com.workshop.demo.payload.ApiResponse;
+import com.workshop.demo.payload.RestaurantResponse;
+import com.workshop.demo.payload.ReviewRequest;
+import com.workshop.demo.repository.RestaurantRepository;
+import com.workshop.demo.repository.ReviewRepository;
+import com.workshop.demo.repository.UserRepository;
+import com.workshop.demo.security.UserPrincipal;
+import com.workshop.demo.service.ReviewService;
+
+public class ReviewServiceImpl implements ReviewService {
+
+    private static final String THIS_REVIEW = " this review";
+
+    private static final String THIS_RESTAURANT = " this restaurant";
+
+    private static final String YOU_DON_T_HAVE_PERMISSION_TO = "You don't have permission to ";
+
+    private static final String ID_STR = "id";
+
+    private static final String REVIEW_STR = "Review";
+
+    private static final String REVIEW_DOES_NOT_BELONG_TO_POSTER = "Review does not belong to poster";
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private ReviewRepository reviewRepository;
+
+    @Autowired
+    private RestaurantRepository restaurantRepository;
+
+    // get all the reviews of the restaurant
+    @Override
+    public Review getAllReviews(String restaurantName) {
+        // Restaurant restaurant = restaurantRepository.findByName(restaurantName)
+        // .orElseThrow(() -> new ResourceNotFoundException(THIS_RESTAURANT, ID_STR,
+        // restaurantName));
+        Review review = reviewRepository.findByRestaurantName(restaurantName)
+                .orElseThrow(() -> new ResourceNotFoundException(REVIEW_STR, ID_STR, restaurantName));
+        if (review.getRestaurant().getName() == restaurantName) {
+            return review;
+        }
+
+        throw new BlogapiException(HttpStatus.BAD_REQUEST, "this restaurant does not have a comment");
+    }
+
+    @Override
+    public Review addReview(ReviewRequest reviewRequest, Long userId, UserPrincipal currentUser) {
+        Restaurant restaurant = restaurantRepository.findByName(reviewRequest.getRestaurantName())
+                .orElseThrow(() -> new ResourceNotFoundException(THIS_RESTAURANT, ID_STR,
+                        reviewRequest.getRestaurantName()));
+        User user = userRepository.getUser(currentUser);
+        Review review = new Review(reviewRequest.getBody());
+        review.setUser(user);
+        review.setRating(reviewRequest.getRating());
+        review.setRestaurant(restaurant);
+        return reviewRepository.save(review);
+    }
+
+    // get the user's specific review with userId and the id of the review
+    @Override
+    public Review getReview(Long userId, Long id) {
```

Comment:
ReviewServiceImpl should override the method defined in ReviewService. The declaration does not match the one in ReviewService.

Reply:
the declaration in ReviewService is changed into        Review getReview(Long userId, Long id);